### PR TITLE
improved reminder feature

### PR DESCRIPTION
**Pull request by ghaith96**
partial Fix for #4
- I displayed reminder time in GoalsPage.
- I also added option to edit reminders in EditPage.
plus some refactoring here and there.
i still need some more info about the last two points:
- Need to allow multiple reminder notifications (newer ones overwrite already existing ones in the notification shade).
- Need to fix time when reminders are displayed.
Anyhow, I can see that the code has a room for improvement, logically and structurally.

### CODE REVIEW

**Comment by urmilshroff** [comment]
Wonderful! Thanks for all your improvements. Here's more info on those last two points:
- If I've set multiple reminders, say one for today afternoon and one for today evening and I don't see my phone throughout the day. The notification will appear for the afternoon one, but the evening one will automatically overwrite the old notification. What should happen is that both notifications stay as separate reminders.
- Basically the whole reminder notification deadline/time is a mess. Sometimes reminders are displayed at the correct day and time, sometimes they're not. It's hard to reproduce since it's so random.

And yeah, the code can be improved quite a bit. It was one of my first proper apps, let alone in Flutter, so there's still a lot of room for learning and improving😅
